django/contrib/gis/db/backends/mysql/schema.py

```diff
@@ -81,3 +95,36 @@ def create_spatial_indexes(self):
                     f"support them.",
                 )
         self.geometry_sql = []
+
+    def _should_have_spatial_index(self, field):
+        # MySQL doesn't support spatial indexes on NULL columns
+        return (
+            isinstance(field, GeometryField) and field.spatial_index and not field.null
+        )
+
+    def _queue_create_spatial_index(self, model, field):
+        """Queues SQL to drop a field's index for create_spatial_indexes()."""
+        qn = self.connection.ops.quote_name
+        db_table = model._meta.db_table
+        self.geometry_sql.append(
+            self.sql_add_spatial_index
+            % {
+                "index": qn(self._create_spatial_index_name(model, field)),
+                "table": qn(db_table),
+                "column": qn(field.column),
+            }
+        )
```

Comment:
I'd rename it to the `_spatial_index_sql()` and add a result to the deferred SQL directly in the `_alter_field()` and `column_sql()`.

Reply:
Fair enough. I'll do the same for Oracle then.